sycl/source/handler.cpp

```diff
@@ -431,11 +431,28 @@ event handler::finalize() {
   // it to the graph to create a node, rather than submit it to the scheduler.
   if (auto GraphImpl = MQueue->getCommandGraph(); GraphImpl) {
     auto EventImpl = std::make_shared<detail::event_impl>();
-
-    // Extract relevant data from the handler and pass to graph to create a
-    // new node representing this command group.
-    std::shared_ptr<ext::oneapi::experimental::detail::node_impl> NodeImpl =
-        GraphImpl->add(MCGType, std::move(CommandGroup));
+    auto NodeImpl =
+        std::make_shared<ext::oneapi::experimental::detail::node_impl>();
```

Comment:
Do we need to actually call `make_shared` here? Since the node created here will be destroyed either way we could just initialize this with `nullptr`.